### ISSUE TITLE
Version::parse and VersionReq::parse accept empty pre-release identifiers

### ISSUE THREAD

**Issue by sarnesjo** [dtolnay/semver#190]
The parse methods on Version and VersionReq both accept empty pre-release identifiers, such as in the version `1.2.3-.`. I'd expect them both to return errors, but they don't:

```rust
Version::parse("1.2.3-.")
// Ok(Version { major: 1, minor: 2, patch: 3, pre: [AlphaNumeric(""), AlphaNumeric("")], build: [] })

VersionReq::parse("1.2.3-.")
// Ok(VersionReq { predicates: [Predicate { op: Compatible, major: 1, minor: Some(2), patch: Some(3), pre: [AlphaNumeric(""), AlphaNumeric("")] }] })
```

That is, the `.` is parsed as two empty AlphaNumeric identifiers.

Variations on this, such as `1.2.3-a.`, `1.2.3-.a`, and `1.2.3-...`, are also accepted.

This is contrary to the [semver spec](https://semver.org/#spec-item-9), which says that "Identifiers MUST NOT be empty".

**Comment by spenserblack** [comment]
Looks like this issue is fixed in this crate's dependency, semver-parser, @ 0.9.0.

**Comment by dtolnay** [comment]
This is fixed in 1.0.0.

```rust
fn main() {
    println!("{}", semver::Version::parse("1.2.3-.").unwrap_err());
    println!("{}", semver::VersionReq::parse("1.2.3-.").unwrap_err());
}
```

```console
empty identifier segment in pre-release identifier
empty identifier segment in pre-release identifier
```
